src/interface_prometheus.py

```diff
@@ -19,7 +19,7 @@ def __init__(self, charm, relation_name, port, path, **job_data):
         super().__init__(charm, relation_name)
         uuid = str(uuid4())
         self.job = {
-            "job_name": f"{relation_name}",
+            "job_name": f"{socket.gethostname()}-{relation_name}",
```

Comment:
Apparently the job_name that is sent to the Prometheus charm needs to be unique as well. By adding the hostname of the instance, it should likely always be unique for different relations.